borg/testsuite/lrucache.py

```diff
@@ -21,20 +21,29 @@ def test(self):
         self.assert_equal(len(c), 2)
         self.assert_equal(c['c'], 2)
         self.assert_equal(c['d'], 3)
-        c['c'] = 22
-        c['e'] = 4
-        self.assert_equal(len(c), 2)
-        self.assert_raises(KeyError, lambda: c['d'])
-        self.assert_equal(c['c'], 22)
-        self.assert_equal(c['e'], 4)
         del c['c']
         self.assert_equal(len(c), 1)
         self.assert_raises(KeyError, lambda: c['c'])
-        self.assert_equal(c['e'], 4)
+        self.assert_equal(c['d'], 3)
+        c.clear()
+        self.assert_equal(c.items(), set())
 
-    def test_pop(self):
-        c = LRUCache(2)
-        c[1] = 1
-        c[2] = 2
-        c.pop(1)
-        c[3] = 3
+    def test_dispose(self):
+        c = LRUCache(2, dispose=lambda f: f.close())
+        f1 = TemporaryFile()
+        f2 = TemporaryFile()
+        f3 = TemporaryFile()
+        c[1] = f1
+        c[2] = f2
+        self.assert_equal(False, f2.closed)
+        c[3] = f3
+        self.assert_equal(False, 1 in c)
+        self.assert_equal(True, f1.closed)
+        self.assert_equal(True, 2 in c)
+        self.assert_equal(False, f2.closed)
+        del c[2]
+        self.assert_equal(False, 2 in c)
+        self.assert_equal(True, f2.closed)
+        c.clear()
+        self.assert_equal(c.items(), set())
+        self.assert_equal(True, f3.closed)
```

Comment:
I've not used either, I'm just cargo-culting with copy+paste here :P.  Since I've re-written lrucache, I guess I could re-write the testcase.  (I.e. that's not likely to cause a problem with merge conflicts).